core/modules/filters/strings.js

```diff
@@ -217,7 +217,10 @@ exports.splitregexp = function(source,operator,options) {
 		return ["RegExp error: " + ex];
 	}
 	source(function(tiddler,title) {
-		Array.prototype.push.apply(result,title.split(regExp));
+		var parts = title.split(regExp).map(function(part){
+			return part+"";	// make sure it's a string
```

Comment:
Hi @pmario I think this should be `return part || ""`.

Reply:
But then `undefined` would return an emtpyString which is inconsistent. Please have a closer look at the OP code examples. IMO `{{{ [<str>splitregexp<re>join[, ]] }}} and {{{ [<str>splitregexp<re>] }}}` should return the same number of elements. This would not be the case, if we use `part || ""`

Reply:
Returning "undefined" as a filter result is user hostile for non-developers and has no precedence among filter operators. On the other hand there is precedence for operators returning an empty string when there is no matching result.

Reply:
I'll change it as requested